src/main/scala/viper/silver/reporter/Reporter.scala

```diff
@@ -163,12 +167,32 @@ case class StdIOReporter(name: String = "stdout_reporter", timeInfo: Boolean = t
         //println( s"Configuration confirmation: $text" )
       case InternalWarningMessage(_) =>        // TODO  use for progress reporting
         //println( s"Internal warning: $text" )
-      case _:SimpleMessage =>
+      case _: SimpleMessage =>
         //println( sm.text )
+      case _: QuantifierInstantiationsMessage => // too verbose, do not print
+      case _: QuantifierChosenTriggersMessage => // too verbose, do not print
+      case t: VerificationTerminationMessage => println(t.toString)
       case _ =>
         println( s"Cannot properly print message of unsupported type: $msg" )
     }
     counter = counter + 1
   }
 }
 
+case class PollingReporter(name: String = "polling_reporter", pass_through_reporter: Reporter) extends Reporter {
+  // this reporter stores the messages it receives and reports them upon polling
+  var messages: Queue[Message] = Queue()
+
+  def report(msg: Message): Unit = this.synchronized {
+    messages = messages.enqueue(msg)
+    pass_through_reporter.report(msg)
+  }
+
+  def getNewMessage(): Message = this.synchronized {
+    return messages.dequeue()
```

Comment:
Drop the `return`, its use is discouraged in Scala (Google will quickly reveal the justification)